src/ParameterLists.f90

```diff
@@ -2743,21 +2743,29 @@ SUBROUTINE verifyTest_Paramtype(thisParam,reqParams,isMatch)
 !> @param reqParams
 !> @param isMatch
 !>
-    SUBROUTINE verifyList_Paramtype(thisParam,reqParams,isMatch)
+    SUBROUTINE verifyList_Paramtype(thisParam,reqParams,e,isMatch)
       CLASS(ParamType),INTENT(INOUT) :: thisParam
       CLASS(ParamType),INTENT(IN) :: reqParams
+      CLASS(ExceptionHandlerType),INTENT(IN) :: e
       LOGICAL(SBK),INTENT(OUT) :: isMatch
       LOGICAL(SBK) :: isValid
+      INTEGER(SIK) :: tmpcount(EXCEPTION_SIZE)
+      TYPE(ExceptionHandlerType) :: tmpe
 
       !Assume the list is valid, check it only if the required parameter
       !list is not empty.
+      tmpe=eParams
+      eParams=e
```

Comment:
I don't think we should modify `eParams` anywhere in this routine.

Reply:
Then why are we passing in an exception handler?

Reply:
So we don't have to modify `eParams` and we can still count/track errors. `e` should be properly setup outside the call to `verifyLists` by the client.